components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -132,10 +133,12 @@ class AtmosphereInput
   //  - layouts: the layout of the vars (used to reshape the views).
   void init(const std::shared_ptr<const grid_type>& grid,
             const std::map<std::string,view_1d_host>& host_views_1d,
-            const std::map<std::string,FieldLayout>&  layouts);
+            const std::map<std::string,FieldLayout>&  layouts,
+            const std::map<std::string, std::string>& dtypes);
 
   // Read fields that were required via parameter list.
   void read_variables ();
+  void read_variables (const int timelevel);
```

Comment:
We can simply have a single method, with a default for timelevel: `void read_variables (const int timelevel = -1);`

Also, can we use `time_index`, rather than `time_level`? The word level makes me think of something with a fixed upper bound (like the vert levels, or the time levels in homme used for RK stages storage).